app/src/main/java/com/example/pygmyhippo/common/RecyclerClickListener.java

```diff
@@ -0,0 +1,13 @@
+package com.example.pygmyhippo.common;
+
+/**
+ * Interface for implementing a click listener for a recycler.
+ * This is generally implemented in the parent fragment of the RecyclerView. You generally need a
+ * reference to the list of data being displayed in the RecyclerView to effectively manipulate it
+ * in onItemClick.
+ *
+ * @see com.example.pygmyhippo.admin.AllEventsFragment
+ */
+public interface RecyclerClickListener {
```

Comment:
I'm curious what this is for?

Reply:
For implementing onclick logic for recyclerview items. You implement it in the parent fragment and when on click happens you should have access to the data and position of the item clicked.